File: java/deepspace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author blanca
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
    
    private final static float EXTRAEFFICIENCY=1.2f;
    private static Dice dice;
    public BetaPowerEfficientSpaceStation(SpaceStation station) {
        super(station);
        dice=new Dice();
    }
    
    public float fire() {
        float normalFire=super.fire(); 
        if(dice.extraEfficiency()) { //ni idea de cual es este fallo
            normalFire *= EXTRAEFFICIENCY;
                         
        }
        return normalFire;
    }
    
}

File: ruby/BetaPowerEfficientSpaceStation.rb
require_relative 'PowerEfficientSpaceStation'
require_relative 'Dice'

module Deepspace

  class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation

    @@EXTRAEFFICIENCY = 1.2

    def initialize(station)
      super
      @dice = Dice.new
    end

    def fire
      f = super
      if(dice.extraEfficiency == true)
        return f*@@EXTRAEFFICIENCY
      end
      return f
    end

    def getUIversion
      BetaPowerEfficientSpaceStationToUI.new(self)
    end

  end # class BetaPowerEfficientStation

end #module
